OSDOCS-7495: add backup and restore (#63608)
openshift/openshift-docs: modules/microshift-restoring-data-backups.adoc

//Module included in the following assemblies:
//
// * microshift_updating/microshift-update-options.adoc

:_content-type: PROCEDURE
[id="microshift-restoring-data-backups-manually_{context}"]
= Restoring application data backups manually

If you are not using an OSTree system, you can restore application data from a backup manually. Backups can be restored after updates, or after other system events that remove or damage required data. Backups are in the `/var/lib/microshift-backups` directory by default. When restoring a backup, you must use the entire file path.

.Prerequisites
* Root access to the host.
* Full path of the data backup file.
* {product-title} is stopped.

[IMPORTANT]
====
{product-title} must be stopped during the backup process to avoid data corruption.
====

.Procedure
. Manually restore an application backup file using the entire file path of the backup you want to restore by using the following command syntax:
+
[source, terminal]
----

> $ sudo microshift restore </var/lib/microshift-backups/my-4.14-backup>

Reviewer: In other places it's `# microshift`, not `$ sudo microshift`

Author: Yeah, I think in another PR we talked about making everything just `$ sudo` because we have mixed use in the Docs. If OK here, I'll just use `$ sudo`.

//Module included in the following assemblies:
//
// * microshift_updating/microshift-update-options.adoc

:_content-type: PROCEDURE
[id="microshift-restoring-data-backups-manually_{context}"]
= Restoring application data backups manually

You can restore application data from a backup manually. Backups can be restored after updates, or after other system events that remove or damage required data. Backups are in the `/var/lib/microshift-backups` directory by default. When restoring a backup, you must use the entire file path.

[NOTE]
====
RPM-OSTree systems back up and restore data automatically.
====

.Prerequisites
* Root access to the host.
* Full path of the data backup file.
* {product-title} is stopped.

.Procedure
. Manually restore an application backup file using the full file path of the backup you want to restore by running the following command:
+
[source,terminal]
----

> $ sudo microshift restore /var/lib/microshift-backups/<my_backup_full_file_path>

Reviewer: To me "full file path" means "full absolute filepath" like `/var/lib/custom-backups/custom`, so interpreting this very literally we get: `/var/lib/microshift-backups/var/lib/custom-backups/custom`
```suggestion
$ sudo microshift restore /var/lib/microshift-backups/<my_backup>
```

//Module included in the following assemblies:
//
// * microshift_updating/microshift-update-options.adoc

:_content-type: PROCEDURE
[id="microshift-restoring-data-backups-manually_{context}"]
= Restoring application data backups manually

You can restore application data from a backup manually. Backups can be restored after updates, or after other system events that remove or damage required data. Backups are in the `/var/lib/microshift-backups` directory by default. When restoring a backup, you must use the entire file path.

[NOTE]
====
RPM-OSTree systems backup and restore data automatically.
====

.Prerequisites
* Root access to the host.
* Full path of the data backup file.
* {product-title} is stopped.

> [IMPORTANT]

Reviewer 2: Should this note be moved to the Backing Up MicroShift data manually section ?

Author: It's there already.

//Module included in the following assemblies:
//
// * microshift_updating/microshift-update-options.adoc

:_content-type: PROCEDURE
[id="microshift-restoring-data-backups-manually_{context}"]
= Restoring application data backups manually

> You can restore application data from a backup manually. Backups can be restored after updates, or after other system events that remove or damage required data. Backups are in the `/var/lib/microshift-backups` directory by default. When restoring a backup, you must use the entire file path.

Reviewer 3: ```suggestion
You can restore application data from a backup manually. Backups can be restored after updates, or after other system events that remove or damage required data. Backups are in the `/var/lib/microshift-backups` directory by default. When you restore a backup, you must use the entire file path.
```
From IBM Style: "When you use a verb phrase that begins with a present participle, such as _creating_, or past participle, such as _created_, at the beginning of a sentence, make sure that the verb phrase modifies the correct word. Failure to do so can result in a dangling modifier."

[NOTE]
====
On an `rpm-ostree` system, {product-title} backs up and restores data automatically.
====

.Prerequisites
* Root access to the host.
* Full path of the data backup file.
* {product-title} is stopped.

.Procedure

> . Manually restore an application backup file using the full file path of the backup you want to restore by running the following command:

Reviewer 3: ```suggestion
. Manually restore an application backup file that uses the full file path of the backup you want to restore by running the following command:
```

??? I0829 07:33:04.719562    6654 run_check.go:28] "Service state" service="microshift-etcd.scope" state="inactive"
??? I0829 07:33:04.719681    6654 data_manager.go:121] "Copying backup to data directory" storage="/var/lib/microshift-backups" name="4.14.0__20230829_113212" data="/var/lib/microshift"
??? I0829 07:33:04.719802    6654 data_manager.go:138] "Renaming existing data dir" data="/var/lib/microshift" renamedTo="/var/lib/microshift.saved"
??? I0829 07:33:04.719913    6654 data_manager.go:186] "Starting copy" cmd="/bin/cp --verbose --recursive --preserve --reflink=auto /var/lib/microshift-backups/4.14.0__20230829_113212 /var/lib/microshift"
??? I0829 07:33:04.725032    6654 data_manager.go:200] "Finished copy" cmd="/bin/cp --verbose --recursive --preserve --reflink=auto /var/lib/microshift-backups/4.14.0__20230829_113212 /var/lib/microshift"
??? I0829 07:33:04.725091    6654 data_manager.go:160] "Removing temporary data directory" path="/var/lib/microshift.saved"
??? I0829 07:33:04.727461    6654 data_manager.go:165] "Copied backup to data directory" name="4.14.0__20230829_113212" data="/var/lib/microshift"
----
+
.Verification

> . Verify that your backup is restored by restarting {product-title} and checking the data.

Reviewer 3: ```suggestion
* Verify that your backup is restored by restarting {product-title} and checking the data.
```
From the [SSG](https://redhat-documentation.github.io/supplementary-style-guide/#single-step-procedures): "When a procedure contains only one step, use an unnumbered bullet."
